### DISCUSSION TITLE
Repository Check (Yana)

### DISCUSSION THREAD

**Discussion by altanborali16** [altanborali16/SWE574-Group2#98] [Q&A]
I am at dev-yana branch 

Run it with docker-compose up

Here are my findings

### Frontend

- Only able to create user from "register" 
- Create user without an email address
- After register page direct me to the home page without a login
- Logged in with created user
- Cannot able to create community
- I created community from swagger
- I saw a subsribe button and cannot access anything
- When clicked on user it did not navigate me to any page

So besides of backend connection there are a lot to change on frontend, which is fine we already created a frontend without backend all data are stored as json on frontend, it also need a lot till the end. 

###  Backend

Checked it from http://localhost:8080/swagger-ui/index.html

I skipped files section, i think that is not critical :)

- POST api/users i think it should create few more fields so in future user can go to user profile page and change them
- GET api/users/{id}/communities & api/communities can have more fields like postCount, memberCount, so it will be easy to show on frontend
- POST api/posts looks fine, can be tricky on frontend but we can talk about it such as userinfo, better field definition or need too much mapping on frontend
- POST api/posts/{id}/upvote it can work if we are only holding the value of points but one user can simply upvote one post multiple times
- POST api/login does not gives an auth token, could not understand how all api's work without auth token
- POST api/communities while creating a community template field is not necessary, when a community created a default template can be created on backend, new templates can be created afterwards.
-  GET api/communities/{id}/members looks fine but it can be in get communit/{id} also isManager field can be added
- GET api/communities/{id}/posts can be converted as single data such as communityWithAllFields, so when user is in the community page, we do not need multiple requests to get everything 
- Cannot find user or userprofile api

I will check schemas tomorrow. But from my rough check I will probably have some troubles with understanding that.

As I mentioned I could not run everything on frontend it can be because of me, so a small talk can help me to understand what I did wrong.

@kotobusdriver

**Comment by kotobusdriver** [comment]
Yes, since I have introduced some changes to the backend, the old frontend now does not work as intended because I did not make the corresponding changes there. 

As to the APIs: 

POST api/users i think it should create few more fields so in future user can go to user profile page and change them : _I could add that but i think it would duplicate information in a new API for "userprofile" (if we choose to have it), see your last point_
GET api/users/{id}/communities & api/communities can have more fields like postCount, memberCount, so it will be easy to show on frontend : _Yes, this can be introduced going forward, looks like just a couple of lines._
POST api/posts looks fine, can be tricky on frontend but we can talk about it such as userinfo, better field definition or need too much mapping on frontend : _Since the frontend now does not work as intended, you could maybe look how it was implemented in the old version which is in my old repo for SWE573_
POST api/posts/{id}/upvote it can work if we are only holding the value of points but one user can simply upvote one post multiple times : _I know about this bug, it needs a new table in the DB with member-post pair_
POST api/login does not gives an auth token, could not understand how all api's work without auth token : _Currently there's no security at all, any user can do anything. This can be added later._
POST api/communities while creating a community template field is not necessary, when a community created a default template can be created on backend, new templates can be created afterwards : _Currently, community owner can create community post templates only during the creation of the community. But a new API and endpoint can be added so that to create additional templates. This will require slight rework._
GET api/communities/{id}/members looks fine but it can be in get communit/{id} also isManager field can be added : _Sure, this could be a nice addition_
GET api/communities/{id}/posts can be converted as single data such as communityWithAllFields, so when user is in the community page, we do not need multiple requests to get everything : _To make it more human readable, yes, good point._
Cannot find user or userprofile API : _That's because there's no such API :)_
